raiden/network/transport/matrix/utils.py

```diff
@@ -163,6 +163,22 @@ def force_user_presence(self, user: User, presence: UserPresence):
         """
         self._userid_to_presence[user.user_id] = presence
 
+    def populate_userids_for_address(self, address: Address, force: bool = False):
+        """ Populate known user ids for the given ``address`` from the server directory.
+
+        If ``force`` is ``True`` perform the directory search even if there
+        already are known users.
+        """
+        if force or not self.get_userids_for_address(address):
+            self.add_userids_for_address(
+                address,
```

Comment:
There's similar code in `start_health_check`, maybe you can refactor this a bit to reduce duplication.

Reply:
Yeah it is but the code over there is about finding peers for communication rooms, so not really in scope for the UserAddressManager. 

The peer finding / room creation code is in some need of refactoring anyway.